core/src/main/java/com/datastax/oss/driver/internal/core/context/DefaultDriverContext.java

```diff
@@ -361,20 +361,23 @@ protected EventBus buildEventBus() {
   @SuppressWarnings("unchecked")
   protected Compressor<ByteBuf> buildCompressor() {
```

Comment:
Suggestion: maybe instead of mocking whole `DriverContext` you can just test this method?
It seems that this method is only using the `DriverExecutionProfile defaultProfile` so you don't need to access anything from `driverContext` besides that. 
But to make it happen you would need to create a second method that is taking `DriverExecutionProfile` as an argument and this method is just delegating to that second method: `buildCompressor(){return buildCompressor(getConfig().getDefaultProfile())}` but I don't know if this is worth the effort.

Reply:
It's an interesting idea @tomekl007 , but I'm reluctant to change the API of DefaultDriverContext to make that happen.  Your point does remind me of something I was wondering, though.

Because of the isolation built into the various components of DefaultDriverContext we probably don't _need_ all the mocks created in MockedDriverContextFactory; we can probably just pass in nulls for most of the dependencies and only do the bits that actually setup the plumbing for the config loader.  I just tested that out locally and it _does_ work.  But given that we already have a working impl now _with_ all the mocks it feels like passing in a bunch of nulls via ProgrammaticArguments would be a step backwards.

Abstracting creation of a mock DefaultDriverContext into the factory leaves open the door for future tests to build upon what's there now to provide different ways of creating these mocks in the future, which seems like something of a goodness.  Hopefully this will enable DefaultDriverContextTest to grow over time as things change in the default context. :crossed_fingers: